app/services/sign_in/logout_redirect_generator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'sign_in/logingov/service'
+
+module SignIn
+  class LogoutRedirectGenerator
+    attr_reader :user, :client_config
+
+    def initialize(user:, client_config:)
+      @user = user
+      @client_config = client_config
+    end
+
+    def perform
+      return unless logout_redirect_uri
+
+      if authenticated_with_logingov?
+        logingov_service.render_logout(encode_logout_redirect)
+      else
+        parse_logout_redirect_uri
+      end
+    end
+
+    private
+
+    def parse_logout_redirect_uri
+      URI.parse(logout_redirect_uri).to_s
+    end
+
+    def authenticated_with_logingov?
+      authenticated_credential == SAML::User::LOGINGOV_CSID
+    end
+
+    def encode_logout_redirect
+      Base64.encode64(generate_logout_state_payload.to_s)
+    end
+
+    def generate_logout_state_payload
+      {
+        logout_redirect: logout_redirect_uri,
+        seed: random_seed
+      }
+    end
+
+    def logout_redirect_uri
+      @logout_redirect_uri ||= client_config.logout_redirect_uri
+    end
+
+    def authenticated_credential
+      @authenticated_credential ||= user.nil? ? nil : user.identity.sign_in[:service_name]
```

Comment:
I'm trying to think how `user.identity` would work here when SiS is split out. Will it have its own version of a `user` and `identity`? Should we pass in the `service_name` instead of a `user`?

Reply:
I think this is reasonable, but a good subject for a different PR.

Ultimately the issue is that the current `authenticate` function that inspects the `access_token` and associates it with a `User` model is very closely catered to `vets-api` needs, but when SiS is separated we'll prob want a completely different `authenticate` function with our own user model (or some analogous model)

Reply:
I'm gonna create an epic for this and we can throw tickets in when we think of things